aten/src/ATen/native/native_functions.yaml

```diff
@@ -46,7 +46,7 @@
   dispatch:
     CUDA: _cudnn_rnn_backward
 
-- func: _cudnn_init_dropout_state(float dropout, bool train, int dropout_seed, *, ScalarType dtype, Layout layout, Device device) -> Tensor
+- func: _cudnn_init_dropout_state(float dropout, bool train, int dropout_seed, *, ScalarType dtype, Layout layout, Device device, bool pin_memory=False) -> Tensor
```

Comment:
This one isn't optional, but the other ones are. What's the difference?

Reply:
Doesn't the fact that these are not optional imply that you might break JIT scripts which mention any of the non-optional variants here?

Reply:
I wish I could do it optional, however native function and jit generators only knows how to wrap all non-optional arguments to TensorOptions or all optional arguments to TensorOptions. There is no easy way to get combination of them (without major rewrite, which makes no sense as we migration off this generation anyway).

Reply:
I think my real question is, why weren't *all* of these optional in the first place?

Reply:
This is part of the bigger `what we do with tensor options` discussion.

Reply:
> Doesn't the fact that these are not optional imply that you might break JIT scripts which mention any of the non-optional variants here?

Answering my question, no, JIT scripts should not be broken, because all occurrences of `pin_memory` are given default arguments.

I think there should still be a Note talking about the inconsistency of `pin_memory` in these sites. Put it in `native_functions.yaml` (or maybe the README in native)

Reply:
There's a lot of discussion about this function when it doesn't even matter (outside of not breaking JIT) -- no one calls this function directly.  So presumably we can just do whatever makes our job easier here (giving everything a default, not using TensorOptions, etc. as long as it doesn't break JIT).

Reply:
That's my fault, I stuck the PR comment on the very first occurrence that had this problem. If you look at the rest of `native_functions.yaml` there are a bunch of other, more public, functions that look this way too. (In the end it's a moot point, I don't believe there is any BC-breakage here, regardless of the status of the function.)